src/modules/group/domain/gruppe.spec.ts

```diff
@@ -0,0 +1,103 @@
+import { faker } from '@faker-js/faker';
+import { Gruppe } from './gruppe.js';
+import {
+    Bildungsziele,
+    Faecherkanon,
+    GruppenTyp,
+    Gruppenbereich,
+    Gruppendifferenzierung,
+    Gruppenoption,
+    Gruppenrollen,
+} from './gruppe.enums.js';
+import { CreateGroupBodyParams } from '../api/create-group.body.params.js';
+import { Jahrgangsstufe } from '../../personenkontext/domain/personenkontext.enums.js';
+import { Laufzeit } from '../persistence/laufzeit.js';
+describe('Gruppe', () => {
+    describe('construct', () => {
+        describe('when constructing a group', () => {
+            it('should return a new group instance', () => {
+                const createGroupBodyParams: CreateGroupBodyParams = {
+                    bezeichnung: faker.lorem.word(),
+                    typ: GruppenTyp.KLASSE,
+                    optionen: [],
+                    differenzierung: Gruppendifferenzierung.E,
+                    bildungsziele: [],
+                    jahrgangsstufen: [],
+                    faecher: [],
+                    referenzgruppen: [],
+                    laufzeit: {},
+                };
+
+                const createdGroup: Gruppe<false> = Gruppe.createGroup(createGroupBodyParams);
+
+                expect(createdGroup.id).toBeUndefined();
+                expect(createdGroup.createdAt).toBeUndefined();
+                expect(createdGroup.updatedAt).toBeUndefined();
+                expect(createdGroup.revision).toBeDefined();
+                expect(createdGroup.mandant).toBe('');
+                expect(createdGroup.organisationId).toBe('');
+                expect(createdGroup.referrer).toEqual(createGroupBodyParams.referrer);
+                expect(createdGroup.bezeichnung).toEqual(createGroupBodyParams.bezeichnung);
+                expect(createdGroup.thema).toEqual(createGroupBodyParams.thema);
+                expect(createdGroup.beschreibung).toEqual(createGroupBodyParams.beschreibung);
+                expect(createdGroup.typ).toEqual(createGroupBodyParams.typ);
+                expect(createdGroup.bereich).toEqual(createGroupBodyParams.bereich);
+                expect(createdGroup.optionen).toEqual(createGroupBodyParams.optionen);
+                expect(createdGroup.differenzierung).toEqual(createGroupBodyParams.differenzierung);
+                expect(createdGroup.bildungsziele).toEqual(createGroupBodyParams.bildungsziele);
+                expect(createdGroup.jahrgangsstufen).toEqual(createGroupBodyParams.jahrgangsstufen);
+                expect(createdGroup.faecher).toEqual(createGroupBodyParams.faecher);
+                expect(createdGroup.referenzgruppen).toEqual(createGroupBodyParams.referenzgruppen);
+                expect(createdGroup.laufzeit).toEqual(createGroupBodyParams.laufzeit);
```

Comment:
For better maintainability and compiler checks:
```suggestion
expect(result).toEqual(expect.objectContaining<Gruppe<false>>({
    id: expect.any(String),
    createdAt: expect.any(Date),
    updated: expect.any(Date),
    ...
}))
```

Reply:
linter will throw an error:   
    id: expect.any(String),
    createdAt: expect.any(Date),
    updated: expect.any(Date),
a possible solution will look like this: 

```suggestion

                expect(createdGroup).toEqual(
                    expect.objectContaining({
                        id: undefined,
                        createdAt: undefined,
                        updatedAt: undefined,
                        revision: '',
                        mandant: '',
                        organisationId: '',
                        referrer: createGroupBodyParams.referrer,
                        thema: createGroupBodyParams.thema,
                        bereich: createGroupBodyParams.bereich,
                        beschreibung: createGroupBodyParams.beschreibung,
                        bezeichnung: createGroupBodyParams.bezeichnung,
                        typ: createGroupBodyParams.typ,
                        optionen: createGroupBodyParams.optionen,
                        differenzierung: createGroupBodyParams.differenzierung,
                        bildungsziele: createGroupBodyParams.bildungsziele,
                        jahrgangsstufen: createGroupBodyParams.jahrgangsstufen,
```